napari/utils/context/_context.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+import sys
+from typing import Any, ChainMap, Dict, Final, MutableMapping, Optional
+from weakref import finalize
+
+from ..events.event import Event, EventEmitter
+
+_null = object()
+
+
+class Context(ChainMap):
+    def __init__(self, *maps: Dict[str, Any]) -> None:
+        super().__init__(*maps)
+        self.changed = EventEmitter(self, 'changed')
+
+    # this requires psygnal or changes to events.py ...
```

Comment:
no, this is a "feature" that i think should eventually be there (really... it would generally be useful of our event emitter).  So it's just a "note" for now. 

https://github.com/tlambert03/psygnal/pull/23